src/configurables/blockchains/loom/index.js

```diff
@@ -108,18 +108,20 @@ export default (function loom () {
 
   const getPrivateKey = () => wallet.privateKey
 
-  const needsWallet = fn => (...args) => {
+  const needsWallet = fn => async (...args) => {
     if (!wallet.address) {
       throw new Error('Wallet does not exist. Please initialize or generate it.')
     }
-    return fn(...args)
+    const result = await fn(...args)
```

Comment:
Does it make any difference?

Reply:
Right now, no, but if you wrap an async function with needWallet.... then you will have problems.
I fixed it here because it happenned to me with Qtum file and it took me long to analyse where the problem came from.

So as I saw you had same potential error, I preferred to fix it right away to avoid surprises